src/tactic/omega/list.lean

```diff
@@ -0,0 +1,248 @@
+  { rw not_lt at h3,
+    rw get_eq_zero_of_le _ h3, apply h1 }
+end 
+
+@[omega] def neg [has_neg α] : list α → list α 
+| []      := [] 
+| (a::as) := -a::(neg as)
+
+@[omega] lemma get_neg [add_group α] : 
+  ∀ {k} {as : list α}, as.neg.get k = -(as.get k) 
+| _ []          := by simp_omega
+| 0 (a::as)     := by simp_omega
+| (k+1) (a::as) := begin simp_omega [get_neg] end
+
+@[omega] lemma length_neg [has_neg α] : 
+  ∀ as : list α, (as).neg.length = as.length 
+| []      := rfl 
+| (a::as) := by simp_omega [list.length, length_neg] 
+
+@[omega] def add [has_add α] : list α → list α → list α 
+| as1 [] := as1
+| [] as2 := as2
+| (a1::as1) (a2::as2) := ((a1+a2)::(add as1 as2))
+
+@[omega] lemma nil_add [has_add α] (as : list α) :
+add [] as = as := by cases as; refl
+
+@[omega] lemma add_nil [has_add α] (as : list α) :
+add as [] = as := by cases as; refl
+
+@[omega] lemma get_add [add_monoid α] : 
+  ∀ {k} {is js : list α}, 
+  (add is js).get k = (is.get k + js.get k) 
+| 0     is      []      := by simp_omega
+| 0     []      js      := by simp_omega
+| 0     (i::is) (j::js) := by simp_omega
+| (k+1) []      js      := by simp_omega
+| (k+1) is      []      := by simp_omega
+| (k+1) (i::is) (j::js) := by simp_omega [get_add]
+
+lemma length_add [has_add α] : 
+  ∀ {is js : list α}, (add is js).length = max is.length js.length 
+| [] js := begin simp_omega, rw max_eq_right, apply zero_le end
+| is [] := begin simp_omega, rw max_eq_left, apply zero_le end
+| (i::is) (j::js) :=
+  by simp_omega [list.length,
+     nat.max_succ_succ, length_add]
+
+lemma map_add_map [has_add α] (f g : α → α) :
+  ∀ {as : list α}, as.map (λ x, f x + g x) = add (as.map f) (as.map g) 
+| []      := rfl
+| (a::as) := begin simp_omega [@map_add_map as], constructor; refl end
+
+@[omega] def sub [has_neg α] [has_sub α] : list α → list α → list α 
+| [] []        := []
+| [] (a2::as2) := neg (a2::as2)
+| (a1::as1) [] := a1::as1
+| (a1::as1) (a2::as2) := (a1-a2)::(sub as1 as2) 
+
+@[omega] lemma nil_sub [has_neg α] [has_sub α] (as : list α) :
+sub [] as = as.neg := by cases as; refl
+
+@[omega] lemma sub_nil [has_neg α] [has_sub α] (as : list α) :
+sub as [] = as := by cases as; refl
+
+@[omega] lemma get_sub [add_group α] : 
+  ∀ {k} {is js : list α}, 
+  (sub is js).get k = (is.get k - js.get k) 
+| 0     is      []      := by simp_omega
+| 0     []      js      := by simp_omega
+| 0     (i::is) (j::js) := by simp_omega
+| (k+1) []      js      := by simp_omega 
+| (k+1) is      []      := by simp_omega 
+| (k+1) (i::is) (j::js) := by simp_omega [get_sub]
+
+lemma length_sub [has_neg α] [has_sub α] : 
+  ∀ {is js : list α}, (sub is js).length = max is.length js.length 
+| [] js := begin simp_omega, rw max_eq_right, apply zero_le end
+| is [] := begin simp_omega, rw max_eq_left, apply zero_le end
+| (i::is) (j::js) :=
+  by simp_omega [list.length,
+     nat.max_succ_succ, length_sub]
+
+def max [inhabited α] [decidable_linear_order α] : list α → α 
```

Comment:
I think this should already exist in mathlib

Reply:
I'm having trouble finding it in mathlib. Is its name something other than `max`?